src/controllers/validator/also-known-as.ts

```diff
@@ -0,0 +1,61 @@
+import type { AlsoKnownAs } from '../../types/shared.js';
+import type { IValidationResult, IValidator, Validatable } from './validator.js';
+
+export class AlsoKnownAsValidator implements IValidator {
+	validateEach(alsoKnownAs: AlsoKnownAs): IValidationResult {
+		if (!alsoKnownAs.uri) {
+			return {
+				valid: false,
+				error: "URI field is required for AlsoKnownAs",
+			};
+		}
+
+		if (typeof alsoKnownAs.uri !== 'string') {
+			return {
+				valid: false,
+				error: "URI field in AlsoKnownAs must be a string",
+			};
+		}
+
+		if (!alsoKnownAs.description) {
+			return {
+				valid: false,
+				error: "Description field is required for AlsoKnownAs",
+			};
+		}
+		if (typeof alsoKnownAs.description !== 'string') {
+			return {
+				valid: false,
+				error: "Description field in AlsoKnownAs must be a string",
```

Comment:
Is this validator only meant to be used with resources? The rules here seem to be highly specific to our implementation of DID-linked resources. `alsoKnownAs` is also a property for DIDs, and this is allowed to be:

1. Just a string (typically a URI string, could be a DID or HTTP URI)
2. An array of strings (could either be DID or HTTP URIs)
3. An array of objects (our style for DID-Linked Resources)